components/script/dom/element.rs

```diff
@@ -1260,6 +1261,20 @@ impl<'dom> LayoutElementHelpers<'dom> for LayoutDom<'dom, Element> {
             })
             .collect()
     }
+
+    #[allow(unsafe_code)]
+    fn get_assigned_slot(&self) -> Option<LayoutDom<'dom, HTMLSlotElement>> {
+        unsafe {
+            self.unsafe_get()
+                .rare_data
+                .borrow_for_layout()
+                .as_ref()?
+                .slottable_data
+                .assigned_slot
+                .as_ref()
+                .map(|sr| sr.to_layout())
```

Comment:
What does `sr` refer to here? How about `assigned_slot_ref`?

Reply:
That's from some other code i copied where it referred to a shadow root. I'll fix it.